sr_robot_launch/src/sr_robot_launch/mock_sr_ur_robot_hw.py

```diff
@@ -0,0 +1,279 @@
+        return self._robot_mode
+
+    def get_safety_mode(self):
+        return self._safety_mode
+
+    def get_program_running(self):
+        return self._program_running
+
+    def get_program_state(self):
+        return self._program_state
+
+    def restart_safety(self):
+        self._set_safety_mode('NORMAL')
+        self._set_robot_mode('POWER_OFF')
+        self._set_program_running(False)
+        self._reboot_required = False
+
+    def power_on(self):
+        if self._e_stop_active:
+            self.emergency_stop()
+            return
+        self._set_safety_mode('NORMAL')
+        self._set_robot_mode('IDLE')
+        self._set_program_running(False)
+
+    def brake_release(self):
+        if self._e_stop_active:
+            self.emergency_stop()
+            return
+        if not self._reboot_required:
+            self._set_safety_mode('NORMAL')
+            self._set_robot_mode('RUNNING')
+            self._set_program_running(False)
+        else:
+            self.fault()
+
+    def unlock_protective_stop(self):
+        self._set_safety_mode('NORMAL')
+        self._set_robot_mode('RUNNING')
+        self._set_program_running(False)
+
+    def power_off(self):
+        self._set_safety_mode('NORMAL')
+        self._set_robot_mode('POWER_OFF')
+        self._set_program_running(False)
+        self._reboot_required = False
+
+    def protective_stop(self):
+        self._set_safety_mode('PROTECTIVE_STOP')
+        self._set_robot_mode('RUNNING')
+        self._set_program_running(False)
+
+    def check_e_stop(self):
+        return self._e_stop_active
+
+    def emergency_stop(self, latch=True):
+        self._e_stop_active = latch
+        if latch:
+            self._set_safety_mode('ROBOT_EMERGENCY_STOP')
+            self._set_robot_mode('IDLE')
+            self._set_program_running(False)
+            self._reboot_required = True
+            self._publish_safety_mode(True)
+        else:
+            self._set_safety_mode('NORMAL')
+            self._set_robot_mode('IDLE')
+            self._set_program_running(False)
+            self._reboot_required = True
+            self._publish_safety_mode(False)
+
+    def resend_robot_program(self):
+        if self._e_stop_active:
+            self.emergency_stop()
+            return
+        if (self.get_safety_mode().safety_mode.mode == SafetyMode.NORMAL and
+                self.get_robot_mode().robot_mode.mode == RobotMode.RUNNING):
+            self._set_program_running(True)
+        else:
+            self._set_program_running(False)
+
+    def fault(self):
+        self._set_safety_mode('FAULT')
+        self._set_robot_mode('STOPPED')
+        self._set_program_running(False)
+
+
+class MockUrRobotHW(object):
+    def __init__(self, side='right'):
+        if 'left' not in side and 'right' not in side:
+            rospy.logerr("side: %s not valid. Valid sides are: 'left, 'right'", side)
+            raise IllegalArgumentError
+            exit(1)
```

Comment:
no need to exit if you raising an exception

Reply:
71d2bc6d3854b4816d8790a2a364fe157085ec51